client/src/modules/fee_center/modals/fee_center.modal.js

```diff
@@ -54,11 +55,24 @@ function FeeCenterModalController($state, FeeCenter, ModalService, Notify, AppCa
             vm.auxiliaryCenter = 1;
           }
           vm.feeCenter.is_cost = reference.is_cost;
-          vm.feeCenter.reference_cost_id = reference.account_reference_id;
-          vm.costCenterReference = {
-            account_reference_id : reference.account_reference_id,
-            is_cost : reference.is_cost,
-          };
+
+          if (reference.is_variable) {
+            vm.feeCenter.reference_cost_variable_id = reference.account_reference_id;
+            vm.variableCostCenterReference = {
+              account_reference_id : reference.account_reference_id,
+              is_cost : reference.is_cost,
+              is_variable : reference.is_variable,
+              is_turnover : reference.is_turnover,
+            };
+          } else {
+            vm.feeCenter.reference_cost_fixed_id = reference.account_reference_id;
+            vm.fixCostCenterReference = {
```

Comment:
It looks like all of these things are the same..  why do we need to attach them to `vm` as different keys?